python/lsst/ip/isr/isrTask.py

```diff
@@ -2155,6 +2216,21 @@ def overscanCorrection(self, ccdExposure, amp):
 
             overscanResults = self.overscan.run(ampImage.getImage(), overscanImage, amp)
 
+            # If we trimmed columns, we need to restore them.
```

Comment:
This hasn't been an issue before because we never cared about the overscan region once the overscan fit was found.  The previous code ignored the trimming, and overscan subtracted only the untrimmed region (leaving the rest alone).  This created square-wave structures, where the image is corrected, a small strip isn't, the bulk of the overscan is corrected as well, and a final small strip isn't.